// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
///////////////////////////////////////////////////////////////////////////////

syntax = "proto3";

package crypto.tink.internal;

message InnerProtoTestProto {
  uint32 inner_proto_uint32_field_1 = 20;
  uint32 inner_proto_uint32_field_2 = 21;
  uint32 inner_proto_uint32_field_3 = 123456;
}

message ProtoTestProto {
  uint32 uint32_field_1 = 1;
  uint32 uint32_field_2 = 2;

  bytes bytes_field_1 = 3;
  bytes bytes_field_2 = 4;

  InnerProtoTestProto inner_proto_field_1 = 10;

  // Maximal tag number is 2^29 - 1 = 536870911.
  uint32 uint32_field_with_large_tag = 536870911;
}
